include/highfive/bits/H5File_misc.hpp

```diff
@@ -76,6 +76,25 @@ inline File::File(const std::string& filename, unsigned openFlags,
     }
 }
 
+inline File::File(hid_t& id){
+    if (id < 0) {
```

Comment:
I would also verify the type of the `hid_t id` here and throw an exception if it's not a file. After this PR, it would be really easy for someone to just move this constructor from `private` to `public`, and there should be definitely a check if the passed `id` is a file (easily forgotten).

Can we make that `const hid_t&`, too?

Reply:
@matz-e I fixed all of this I think. 
Actually now I think it is possible to make this private `File` constructor public and do the same for every object, how do you think? It would allow to check whether object is `File`, `Dataset` or anything and  allow users that partly use original hdf5 library cooperate with HighFive.